src/subscript-loc.c

```diff
@@ -142,27 +142,32 @@ r_obj* int_as_location(r_obj* subscript,
       if (opts->missing == SUBSCRIPT_MISSING_ERROR) {
         stop_subscript_missing(subscript, opts);
       }
-    } else {
-      if (elt < 0) {
-        switch (opts->loc_negative) {
-        case LOC_NEGATIVE_INVERT: return int_invert_location(subscript, n, opts);
-        case LOC_NEGATIVE_ERROR: stop_location_negative(subscript, opts);
-        case LOC_NEGATIVE_IGNORE: break;
-        }
+    } else if (elt == 0) {
```

Comment:
Nothing changed with the 0 case, I just lifted it up to be the 2nd condition here (after `NA`)